vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -17,6 +17,19 @@ class Engine < ::Rails::Engine
         })
       end
 
+      LoginController.class_eval do
+        before_action :page_job, only: [:public, :private]
+
+        :private
+
+        def page_job
+          if request.referer !~ /github\.com/
```

Comment:
I attempted to filter out by OAuth params `code` & `state`, but theres actually 3 calls on `/login/scope` during the process of authentication - 2 of which have no OAuth params which creates duplicate pages.